BookReader/BookReader.js

```diff
@@ -1430,13 +1418,26 @@ BookReader.prototype.nextReduce = function( currentReduce, direction, reductionF
             }
         }
         return reductionFactors[newReduceIndex];
-    }
-
-    // Asked for specific autofit mode
-    for (var i = 0; i < reductionFactors.length; i++) {
-        if (reductionFactors[i].autofit == direction) {
-            return reductionFactors[i];
-        }
+    } else if (direction === 'auto') {
+      // Auto mode chooses the least reduction
+      var choice = null;
+      for (var i = 0; i < reductionFactors.length; i++) {
```

Comment:
Suggestion: Same hoisting issue as above. Also see the similar `for` loop below.